File: PkmnTeamBuilder/PkmnTeamBuilder.Api/Controllers/Team/TeamService.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace PkmnTeamBuilder.Api.Controllers.Team
{
    public interface ITeamService
    {
        TeamModel GetTeam(string code);
        TeamModel AddTeam(TeamModel model);
        TeamModel UpdateTeam(TeamModel model);
        IEnumerable<TeamModel> GetAllTeams(int skip, int take, int filterType, string search);
        IEnumerable<TeamModel> GetMyTeams(string userId);
        void DeleteTeam(int id);
        void LinkTeam(string code, string userId);

        LikeModel UpdateLike(int teamId, string userId);
    }

    public class TeamService : ITeamService
    {
        ITeamRepository _repo;

        public TeamService(ITeamRepository repo)
        {
            _repo = repo;
        }

        public TeamModel GetTeam(string code)
        {
            return _repo.GetTeam(code);
        }

        public TeamModel AddTeam(TeamModel model)
        {
            return _repo.AddTeam(model);
        }

        public TeamModel UpdateTeam(TeamModel model)
        {
            return _repo.UpdateTeam(model);
        }

        public IEnumerable<TeamModel> GetAllTeams(int skip, int take, int filterType, string search)
        {
            return _repo.GetAllTeams(skip, take, filterType, search);
        }

        public IEnumerable<TeamModel> GetMyTeams(string userId)
        {
            return _repo.GetMyTeams(userId);
        }

        public void DeleteTeam(int id)
        {
            _repo.DeleteTeam(id);
        }

        public void LinkTeam(string code, string userId)
        {
            _repo.LinkTeam(code, userId);
        }

        public LikeModel UpdateLike(int teamId, string userId)
        {
            return _repo.UpdateLike(teamId, userId);
        }
    }
}

File: PkmnTeamBuilder/PkmnTeamBuilder/src/app/shared/services/team.service.ts
import { Injectable } from "@angular/core";
import { BaseService } from "./base.service";
import { HttpClient } from "@angular/common/http";
import { ConfigService } from "./config.service";
import { Team } from "../models";
import { Observable } from 'rxjs';

@Injectable()
export class TeamService extends BaseService {
    baseUrl = '';

    constructor(
        private http: HttpClient,
        private config: ConfigService) {
        super();

        this.baseUrl = config.getApiURL();
    }

    public saveTeam(team: Team): Observable<any> {
        const userId = localStorage.getItem('userId');

        team.userId = userId;

        team.members.forEach((value, index, array) => {
            value.userId = userId;
        });

        return this.http.post(this.baseUrl + '/team/', team);
    }

    public updateTeam(team: Team) {
        const userId = localStorage.getItem('userId');

        team.userId = userId;

        team.members.forEach((value, index, array) => {
            value.userId = userId;
        });
        
        return this.http.put(this.baseUrl + `/team/`, team);
    }

    public getAllTeams(skip: number, take: number) {
        return this.http.get(this.baseUrl + `/team?skip=${skip}&take=${take}`);
    }

    public getTeamsForUser(skip: number, take: number, search: string) {
        return this.http.get(this.baseUrl + `/team?skip=${skip}&take=${take}&filterType=7&search=${search}`);
    }

    public getFilteredTeams(skip: number, take: number, filterType: number, search: string) {
        return this.http.get(this.baseUrl +
            `/team?skip=${skip}&take=${take}&filterType=${filterType}&search=${search}`
        );
    }

    public getMyTeams(userId: string) {
        return this.http.get(this.baseUrl + `/team/user/${userId}`);
    }

    public getTeam(code: string) {
        return this.http.get(this.baseUrl + `/team/code/${code}`);
    }

    public deleteTeam(id: number) {
        return this.http.delete(this.baseUrl + `/team/${id}`);
    }

    public linkTeam(code: string, userId: string) {
        return this.http.patch(this.baseUrl + `/team/${code}/link/${userId}`, null);
    }
}